blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -134,29 +134,29 @@ builder.addGlobalStore(
 
-But keep in mind, all store operations (also get) will lead to deserialization that costs you some CPU.
+Remember that all store operations (like get) will lead to deserialization that costs you some CPU.
 
 
 ## Many source topics within a stream
-At first, we designed [Kestra](https://github.com/kestra-io/kestra) to have only one **huge** stream for all the processing from the executor. At first, it seems to be cool but this lead to some drawback.
+At first, we designed Kestra to have only one **huge** stream for all the processing from the executor. At first, it seemed cool, but this led to some drawbacks.
 
 
-Here is the last version of our main and only one Kafka Stream with many topics 🙉:
+Here is the last version of our main and only Kafka Stream with many topics 🙉:
 ![Kestra Topology](./2022-06-28-tips-kafka-streams-distributed/topology.jpg)
-Yes, this is some kind of very large Kafka Streams. It was working almost despite the complexity of this one. But the major drawback was :
-- **Monitoring**: all the metrics are under the same consumer groups
-- **Debugging**: during a crash, each topic is consumed independently, when a message failed, the whole process is crashed
-- and the most important is **lag**: since Kafka Stream optimizes the consumption of messages by themselves, a topic with large outputs could lead to lag on other topics unrelated topics. In that case, impossible to understand properly the lag on our consumer.
+Yes, this is a huge Kafka Stream. It was working almost despite the complexity of this one. But the major drawbacks were :
+- **Monitoring**: All the metrics are under the same consumer groups
+- **Debugging**: Each topic is consumed independently during a crash. When a message fails, the whole process crashes.
+- **Lag**: This is the most important one. Since Kafka Streams optimize the consumption of messages by themselves, a topic with large outputs could lead to lag on unrelated topics. In that case, it is impossible to understand the lag on our consumers.
 
-Now, we have decided to split into multiple [streams](https://github.com/kestra-io/kestra/tree/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/executors) to be able to monitors and understand properly the lag on our Kafka Streams.
+Now, we have decided to split into multiple [streams](https://github.com/kestra-io/kestra/tree/develop/runner-kafka/src/main/java/io/kestra/runner/kafka/executors) to be able to monitor and properly understand the lag on our Kafka Streams.
 
-How to split your evil streams? We simply choose to consume only one time a topic (to avoid large network transit), so we grouped all streams by source topics.
+How to split your evil streams? We chose to consume only one time a topic (to avoid large network transit), so we grouped all streams by source topics.
```

Comment:
What are evil streams?

Reply:
Evil stream in my mind was the giant stream with a lot of topics and a lot of task inside that is visible on the Topology page. 
It's hard to maintain and have a lot of performance issues